### PR TITLE
Data source for AppConnector and AppGateway resources

### PR DESCRIPTION
**Pull request by palramanathan**
Data source for AppConnector and AppGateway resources

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
`google_beyondcorp_app_connector`
```
```release-note:new-datasource
`google_beyondcorp_app_gateway`
```

### CODE REVIEW

**Comment by shuyama1** [comment]
Sorry, I'm not able to find time to review this PR today. Assigning a new reviewer, since I'll be out a week and half.

File: mmv1/third_party/terraform/tests/data_source_google_beyondcorp_app_connector_test.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	return Nprintf(`
+resource "google_service_account" "service_account" {
+	account_id   = "tf-test-my-account%{random_suffix}"
+	display_name = "Test Service Account"
+}
+
+resource "google_beyondcorp_app_connector" "foo" {
+  	name     = "tf-test-appconnector-%{random_suffix}"
+	principal_info {
+	service_account {
```
Comment (roaks3) [inline mmv1/third_party/terraform/tests/data_source_google_beyondcorp_app_connector_test.go:107]: Nit: Would you mind cleaning up the indentation to improve readability of this config?

**Review by roaks3 — APPROVED** [review]
One spacing nit, but otherwise LGTM.

I'll add that it may be to your benefit to split this into 2 PRs, one for each data source, so that if something goes wrong we only have to revert one of them. Up to you though.

**Comment by palramanathan** [comment]
Thanks, will update spacing in the PR shortly. Let me keep this as 1 PR since reverting both is ok. Going forward have 1 resource/data source per PR.

File: mmv1/third_party/terraform/tests/data_source_google_beyondcorp_app_connector_test.go (hunk 1 above)
Comment (palramanathan) [reply mmv1/third_party/terraform/tests/data_source_google_beyondcorp_app_connector_test.go]: Done

**Pull request merged**